main.py

```diff
@@ -1,33 +1,79 @@
+        user = update.effective_user
+        await context.bot.send_message(
+            chat_id=update.effective_chat.id,
+            text=f"Hello, {user.first_name}! Welcome to our Telegram bot."
+        )
+        context.user_data["has_started"] = True
+        loguru_logger.info("Starting the Telegram bot")
+
+    async def calculate(self, update: Update, context: ContextTypes.DEFAULT_TYPE):
+        result = self.UNPROCESSABLE_CONTEXT_MSG
+        if context.user_data.get("has_started"):
+            text = update.message.text
+            loguru_logger.info(f"Received text: {text}")
+
+            keyword_pos = text.find(TGBotCommands.CALCULATE.value)
+            expression = text[keyword_pos + len(TGBotCommands.CALCULATE.value):].strip()
+            result = self._calculate_expression(expression)
 
-# Load the .env file to read environment variables
-load_dotenv()
+        await context.bot.send_message(
+            chat_id=update.effective_chat.id,
+            text=result,
+            parse_mode="MarkdownV2"
+        )
 
-# Get the Telegram bot token from the environment variable
-telegram_token = os.getenv('TELEGRAM_TOKEN')
+    def _calculate_expression(self, expression: str) -> str:
+        expression = expression.replace("plus", "+").replace("minus", "-")
+        valid_expression = "".join(re.findall(r"[\d]+|[-+]", expression))
+        try:
+            result = eval(valid_expression)
+            return f"Calculated with result: {result}"
+        except Exception:
+            return self.UNPROCESSABLE_CONTEXT_MSG
 
-# Define a command handler for the '/start' command
-async def start(update: Update, context: ContextTypes.DEFAULT_TYPE):
-    user = update.effective_user
-    await context.bot.send_message(
-        chat_id=update.effective_chat.id,
-        text=f"Hello, {user.first_name}! Welcome to our Telegram bot."
-    )
+    async def handle_message(self, update: Update, context: ContextTypes.DEFAULT_TYPE):
+        text = update.message.text
 
-# Define the main function to run the Telegram bot
-def main():
-    # Initialize the Telegram bot with the token
-    application = Application.builder().token(telegram_token).build()
+        for keyword in TGBotCommands:
+            command = keyword.value
+            if command in text:
+                await getattr(self, command)(update, context)
+                break
+        else:
+            await context.bot.send_message(
+                chat_id=update.effective_chat.id,
+                text=self.UNPROCESSABLE_CONTEXT_MSG,
+                parse_mode="MarkdownV2"
+            )
 
-    # Add a handler for the '/start' command
-    application.add_handler(CommandHandler('start', start))
+    def run(self):
+        self.application.run_polling()
 
-    # Start the bot to poll for messages
-    application.run_polling()
 
-# Entry point for the script
 if __name__ == "__main__":
-    main()
+    bot = TelegramBot(config.TELEGRAM_TOKEN)
+    bot.run()
```

Comment:
make main() where you run application and wrap it with try-except, so logger can write down various errors, which could be handled later.